download_from_csv.py

```diff
@@ -86,20 +188,15 @@ def download_finish(d):
         print '[\033[93mSkipping\033[00m] %s by %s' % \
             (song['name'], song['artist'])
         continue
-    opts = {
-        'format': 'bestaudio/best',
-        'forcejson': True,
-        'postprocessors': [{
-            'key': 'FFmpegExtractAudio',
-            'preferredcodec': 'mp3',
-            'preferredquality': '256',
-        }],
-        'progress_hooks': [download_finish],
-        'logger': MyLogger(),
-        'outtmpl': folder + '/' + song['name'] + ' - ' + song['artist'] + '.%(ext)s'
-    }
-    url = ' '.join([song['name'], song['artist'], 'audio', 'youtube'])
-    url = 'gvsearch1:' + url
+    opts['progress_hooks'] = [download_finish]
+    opts['logger'] = MyLogger()
+    opts['outtmpl'] = folder + '/' + song['name']\
+        + ' - ' + song['artist'] + '.%(ext)s'
+    url = ' '.join([song['name'], song['artist']])
+    if args.search:
```

Comment:
I tried making all the changes you suggested. Kindly, take a look at the latest commit and let me know if I missed something. Thanks for some great pieces of advice above.